packages/core/core/src/registerCoreWithSerializer.js

```diff
@@ -0,0 +1,31 @@
+// @flow strict-local
+
+import {registerSerializableClass} from '@parcel/utils';
+import Asset from './Asset';
+import AssetGraph from './AssetGraph';
+import Dependency from './Dependency';
+import Environment from './Environment';
+// $FlowFixMe this is untyped
+import packageJson from '../package.json';
+
+const packageVersion = packageJson.version;
+if (typeof packageVersion !== 'string') {
+  throw new Error('Expected package version to be a string');
+}
+
+let registered;
+export default function registerCoreWithSerializer() {
+  if (registered) {
+    return;
+  }
+
+  register('Asset', Asset);
```

Comment:
I created this list based on logging requests to the serializer for object values. Apparently this is all that was needed?

Not sure how to make sure that it's clear that if you add a new class that must be serialized, it doesn't slip through the cracks, since we don't throw on unexpected constructors.

Maybe we should?

Reply:
what do you think about having these register calls inside the files where the types are declared rather than maintaining a list here?

Reply:
I guess that would require us to actually `import` all of those types before tying to serialize anything though... maybe this is clearer then. 🤷‍♂

Reply:
This was my exact thought process 😅